### PR TITLE
Fix for the Balance in group problem

### PR DESCRIPTION
**Pull request by chime-mu**
This refers to the bug "Available funds and Group balance values are wrong ".

The _Balance in group number_ at $1527 is actually correct. The problem was that buckets with `status == 'funded'` and `archived_at` not NULL was displayed as cancelled - which they weren't.

I've tested this with the database from September 9, when Jessy discovered the original bug and this fix solves the problem by displaying the funded buckets correctly. The sum of the five buckets in Funded plus the $168 (which is a different problem) is $1527. See attached image.

I've changed the code so it's:
1. More readable
2. Easier to verify it's correct.

skaermbillede 2017-10-24 kl 20 10 22

Before deploying this fix we should consider if we need to notify all admins who has buckets suddenly jumping from Cancelled to Funded.

### CODE REVIEW

**Comment by chime-mu** [comment]
Hi @jlopker - can I persuade you to take a look at this?

File: app/models/bucket-model.coffee [hunk 1]
```diff
@@ -51,5 +51,21 @@ global.cobudgetApp.factory 'BucketModel', (BaseModel) ->
...
+      @status == 'live' && !@archivedAt
+
+    isFunded: ->
+      @status == 'funded' && !@paidAt
+
+    isComplete: ->
+      !!@paidAt && @status == 'funded'
+
+    isCancelled: ->
+      !!@archivedAt && !@paidAt && @status != 'funded'
```
Comment (jlopker) [inline app/models/bucket-model.coffee]: hmm i'm pretty sure that a cancelled bucket can also have a status of funded..

Comment (chime-mu) [reply app/models/bucket-model.coffee]: I did an analysis of the old code base and the current base to see how all these state variables could evolve.

Take a look at [the old state model](https://github.com/cobudget/cobudget-api/wiki/Old-bucket-state-model) and [the new state model](https://github.com/cobudget/cobudget-api/wiki/Current-bucket-state-model). I'm not saying these are correct, so happy to told where they're wrong.

Could you find out how a bucket could be cancelled and still have `status == 'funded'`? 

I think we should add it to the documentation first and then change the code.

**Comment by jlopker** [comment]
@chime-mu this looks good to go for me!

**Pull request merged**